### PR TITLE
ci: add trigger workflow for updating git submodules

### PR DESCRIPTION
**Pull request by maxrake**
This new workflow is meant to be triggered by each of the `phylum-dev` repositories that are included in the `documentation` repository as git submodules. The trigger will happen during the release process for the external repo and will include the tag for the release.

### CODE REVIEW

File: .github/workflows/trigger.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        continue-on-error: true
+        run: |
+          git config user.name 'phylum-bot'
+          git config user.email '69485888+phylum-bot@users.noreply.github.com'
+          git add "ext/${REPO_NAME}"
+          git commit -m "build: update \`${REPO_NAME}\` submodule to \`${TAG_NAME}\`"
+          git push --force origin "HEAD:update-${REPO_NAME}-submodule"
+
+      - name: Create Pull Request
+        if: ${{ steps.commit.outcome == 'success' }}
```
Comment (kylewillmon) [inline .github/workflows/trigger.yml:57]: The `continue-on-error` logic seems unnecessary here. Errors are not expected, right?

```suggestion
      - name: Commit changes
        run: |
          git config user.name 'phylum-bot'
          git config user.email '69485888+phylum-bot@users.noreply.github.com'
          git add "ext/${REPO_NAME}"
          git commit -m "build: update \`${REPO_NAME}\` submodule to \`${TAG_NAME}\`"
          git push --force origin "HEAD:update-${REPO_NAME}-submodule"

      - name: Create Pull Request
```

File: .github/workflows/trigger.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          github-token: ${{ secrets.GH_RELEASE_PAT }}
+          script: |
+            var repo = process.env.REPO_NAME;
+            var tag = process.env.TAG_NAME;
```
Comment (kylewillmon) [inline .github/workflows/trigger.yml:63]: Very minor, but we should prefer `const` (or `let`) over `var`

```suggestion
            const repo = process.env.REPO_NAME;
            const tag = process.env.TAG_NAME;
```

**Review by kylewillmon — CHANGES_REQUESTED** [review]

File: .github/workflows/trigger.yml
Comment (kylewillmon) [inline .github/workflows/trigger.yml:1]: `trigger.yml` doesn't seem like a very descriptive name for this file. I'd prefer `update-submodule.yml`

File: .github/workflows/trigger.yml (hunk 1 above)
Comment (maxrake) [reply .github/workflows/trigger.yml]: > Errors are not expected, right?

I believe that is correct. It has been updated.

**Review by kylewillmon — DISMISSED** [review]

File: .github/workflows/trigger.yml (hunk 2 above)
Comment (maxrake) [reply .github/workflows/trigger.yml]: changed

File: .github/workflows/trigger.yml
Comment (maxrake) [reply .github/workflows/trigger.yml:1]: changed

**Review by kylewillmon — APPROVED** [review]

File: .github/workflows/trigger.yml
Comment (kylewillmon) [reply .github/workflows/trigger.yml:1]: With an underscore... Gross, but at least it's descriptive.

Comment (maxrake) [reply .github/workflows/trigger.yml:1]: LOL...I think underscores are the better looking option. Plus, there is already the `auto_updates.yml` workflow name to match style (which I also named!)

**Pull request merged**